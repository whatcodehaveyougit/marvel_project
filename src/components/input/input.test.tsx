import { render, screen, fireEvent } from "@testing-library/react";
import CustomInput from "./input.component";
import { describe, it, expect } from "@jest/globals";

describe("CustomInput", () => {
  it("renders correctly", () => {
    const tree = render(<CustomInput />).asFragment();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly with props", () => {
    const tree = render(
      <CustomInput type="text" name="test" placeholder="test" />
    ).asFragment();
    expect(tree).toMatchSnapshot();
  });

  it("user types into input", () => {
    render(<CustomInput type="text" name="test" placeholder="test" />);
    const input: TestElement | null = screen.queryByTestId("custom-input");
    if (input) {
      fireEvent.change(input, { target: { value: "123" } });
    }
    expect(hasInputValue(input, "123")).toBe(true);
  });
});

type TestElement = Document | Element | Window | Node;

function hasInputValue(e: TestElement | null, inputValue: string) {
  if (!e) return false;
  return screen.getByDisplayValue(inputValue) === e;
}
