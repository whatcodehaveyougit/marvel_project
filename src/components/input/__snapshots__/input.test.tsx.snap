// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomInput renders correctly 1`] = `
<DocumentFragment>
  <input
    data-testid="custom-input"
    type="text"
  />
</DocumentFragment>
`;

exports[`CustomInput renders correctly with props 1`] = `
<DocumentFragment>
  <input
    data-testid="custom-input"
    name="test"
    placeholder="test"
    type="text"
  />
</DocumentFragment>
`;
